File: src/main/java/me/redth/mmcutils/HeightLimitHud.java
package me.redth.mmcutils;

import cc.polyfrost.oneconfig.hud.SingleTextHud;
import net.minecraft.client.Minecraft;

public class HeightLimitHud extends SingleTextHud {
    public HeightLimitHud() {
        super("Height Limit Distance", false);
    }

    @Override
    protected String getText(boolean example) {
        if (!MMCUtils.inBridgingGame) return "3";
        int distance = 100 - (int) Minecraft.getMinecraft().thePlayer.posY;
        return String.valueOf(distance);
    }

    @Override
    protected boolean shouldShow() {
        return super.shouldShow() && MMCUtils.inBridgingGame;
    }
}

File: src/main/kotlin/me/redth/mmcutils/config/HeightLimitHud.kt
package me.redth.mmcutils.config

import cc.polyfrost.oneconfig.hud.SingleTextHud
import cc.polyfrost.oneconfig.utils.dsl.mc
import me.redth.mmcutils.core.Core

class HeightLimitHud : SingleTextHud("Height Limit Distance", false) {
    override fun getText(example: Boolean) = if (Core.inBridgingGame) {
        val distance = 100 - mc.thePlayer.posY.toInt()
        distance.toString()
    } else "3"

    override fun shouldShow() = super.shouldShow() && Core.inBridgingGame
}